pkg/controllers/job/job_controller.go

```diff
@@ -312,11 +321,15 @@ func (cc *Controller) processNextReq(count uint32) bool {
 	}
 
 	if err := st.Execute(action); err != nil {
-		glog.Errorf("Failed to handle Job <%s/%s>: %v",
-			jobInfo.Job.Namespace, jobInfo.Job.Name, err)
-		// If any error, requeue it.
-		queue.AddRateLimited(req)
-		return true
+		if queue.NumRequeues(req) < maxRetries {
+			glog.V(2).Infof("Failed to handle Job <%s/%s>: %v",
+				jobInfo.Job.Namespace, jobInfo.Job.Name, err)
+			// If any error, requeue it.
+			queue.AddRateLimited(req)
+			return true
+		}
+
+		glog.V(2).Infof("Dropping job<%s/%s> out of the queue: %v", jobInfo.Job.Namespace, jobInfo.Job.Name, err)
```

Comment:
``glog.V(2).Warnf`` ?, also better to announce in log that its because of maxRetries exceeded.